app_backend/github_client.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env -S uv run
+
+import logging
+from typing import Dict, List, Tuple
+
+import httpx
+
+from app_backend.models import GitCommit
+
+logger = logging.getLogger(__name__)
+
+
+class GitHubClient:
```

Comment:
Add env variables for forced disabling of git and github clients. Amend PR description with instructions on how to test it, forcing the server to use either git or github respectively (usually on a dev machine both would work, so it would always default to the local git client).